vernac/g_vernac.mlg

```diff
@@ -236,6 +236,7 @@ GRAMMAR EXTEND Gram
       | IDENT "Let"; "CoFixpoint"; corecs = LIST1 cofix_definition SEP "with" ->
           { VernacCoFixpoint (DoDischarge, corecs) }
       | IDENT "Scheme"; l = LIST1 scheme SEP "with" -> { VernacScheme l }
+      | IDENT "Scheme"; IDENT "Equality"; IDENT "for" ; id = smart_global -> { VernacSchemeEquality id }
```

Comment:
This differs from the old syntax, which let you define multiple induction principles in a single command using the "with" clause.  This doesn't even allow defining 2 Equality principles with the new command, let alone a single command that defines, say, both Equality and Induction principles.  I don't know if that would be taking away important functionality.  The even_odd/odd_even example in this section suggests that defining 2 principles in one command matters.

I expect we'd want to deprecate the forms that are no longer supported in the next release (N) and then remove them and change the grammar in release N+1.  It can be cumbersome for developers but we don't like to abruptly break user code.

![image](https://user-images.githubusercontent.com/1253341/128280421-32c90d3f-11c7-48f2-aba5-71ee8af4bbd1.png)

Reply:
It wasn't even possible to define 2 Equality principles with the previous syntax, nor was it possible to name them nor mix them with other schemes. The with syntax only really makes sense for the `Induction`/`Minimality`/`Elimination`/`Case` schemes, where it is really about mutual inductive types.

The Equality scheme had many checks preventing the with clause, and there was code to separate Equality schemes from a list of schemes. So far, people seem to agree that they were bundled together simply to share the vernac command. They do not really have any semantic overlap.

This new syntax is a much more faithful representation of what is actually going on in that Equality is a completely different command which has very little to do with the other induction schemes.

Reply:
OK, if the old syntax didn't work for Equality then it would not be a compatibility issue.